### PR TITLE
Update feed to load ports in parallel

### PR DESCRIPTION
**Pull request by lsjroberts**
This isn't perfect, there are still some errors being thrown in the console from ports that couldn't be loaded. But the feed is no longer blocked.

### CODE REVIEW

File: scripts/feed.js [hunk 1]
```diff
@@ -14,11 +14,17 @@ function Feed(feed_urls)
 
     for(id in this.feed_urls){
       var archive = new DatArchive(this.feed_urls[id]);
-      var fileEvents = archive.createFileActivityStream()
+      try {
+        fileEvents = archive.createFileActivityStream();
```
Comment (lsjroberts) [inline scripts/feed.js:18]: For some reason even with a try catch around this it still throws an uncaught error. Maybe I'm doing something wrong?

File: scripts/feed.js [hunk 2]
```diff
@@ -32,39 +38,53 @@ function Feed(feed_urls)
...
+          r.portal.port_status_el.innerHTML = (online_ports_count - 1)+" online";
+
+          entries = entries.concat(feed_entries);
+
+          // Sort
+          var sorted_entries = entries.sort(function(a,b){
+            return a.timestamp < b.timestamp ? -1 : 1;
+          });
+
+          this.refresh(sorted_entries.reverse());
```
Comment (lsjroberts) [inline scripts/feed.js:59]: It sorts and re-renders every time a port is loaded, perhaps not the most efficient thing but it should be ok for now.

File: scripts/feed.js (hunk 1 above)
Comment (neauoire) [reply scripts/feed.js:18]: Could it be an async issue with catch? I think archive.createActivityStream is mean to be into async

File: scripts/feed.js [hunk 3]
```diff
@@ -32,39 +38,53 @@ function Feed(feed_urls)
...
-
-    var sorted_entries = entries.sort(function(a,b){
-       return a.timestamp < b.timestamp ? -1 : 1;
+    var online_ports_count = 0;
+
+    var archive_promises = this.archives.map((archive) => (
+      this.get_feed(archive)
+        .then((feed_entries) => {
+          online_ports_count += 1;
+          r.portal.port_status_el.innerHTML = (online_ports_count - 1)+" online";
```
Comment (lsjroberts) [inline scripts/feed.js:50]: screen shot 2017-10-13 at 23 12 31

File: scripts/feed.js (hunk 1 above)
Comment (lsjroberts) [reply scripts/feed.js:18]: Ah good point, probably

**Comment by lsjroberts** [comment]
Fixes #8

**Comment by neauoire** [comment]
It works really well! thank you

**Pull request merged**